src/thunderbots/software/ai/navigator/path_planner/path_planner.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+#include <ai/world/world.h>
+
+#include <optional>
+#include <vector>
+
+#include "geom/point.h"
+
+class PathPlanner
+{
+   public:
+    /**
+     * Finds a path from start to dest if it exists, otherwise return std::nullopt.
+     * @param start the start point.
+     * @param dest the destination point.
+     * @return a path from start to dest if it exists, otherwise std::nullopt.
+     */
+    virtual std::optional<std::vector<Point>> findPath(const Point &start,
```

Comment:
Some implementations of path planners will require obstacles and/or violation functions. In the case of Theta*, we would use obstacles for graph vertex generation and a violation function for field boundaries. We should provide both of these for the path planner to consume.

Reply:
The violation could be more generic as well. For example if we want to negatively weight certain areas of the field.

Reply:
Would A* use violation functions for robots too? Can theta* use boundary polygons for field boundaries?

Reply:
Yup, at the most abstract level all we need for theta* or A* is a function that tells us if a given point is within an obstacle or not, and a function that gives us a weight for a given point.